File: 0451-sort-characters-by-frequency/0451-sort-characters-by-frequency.java

class Solution {
    public String frequencySort(String s) {
        Map<Character, Integer> m = new HashMap<>();
        for (char i : s.toCharArray()) {
            m.put(i, m.getOrDefault(i, 0) + 1);
        }
        PriorityQueue<Map.Entry<Character, Integer>> pq = new PriorityQueue<>(
            (a, b) -> b.getValue() - a.getValue()
        );
        pq.addAll(m.entrySet());
        StringBuilder ans = new StringBuilder();
        while (!pq.isEmpty()) {
            Map.Entry<Character, Integer> entry = pq.poll();
            for (int i = 0; i < entry.getValue(); i++) {
                ans.append(entry.getKey());
            }
        }
        return ans.toString();
    }
}

File: 0451-sort-characters-by-frequency/0451-sort-characters-by-frequency.cpp
class Solution {
public:
    string frequencySort(string s) {
        unordered_map<char,int>m;
        for(auto i: s)m[i]++;
        priority_queue<pair<int,char>>pq;
        for(auto i: m)pq.push({i.second,i.first});
        string ans="";
        while(!pq.empty()){
            for(int i=0;i<pq.top().first;i++)ans+=pq.top().second;
            pq.pop();
        }
        return ans;
    }
};